frontend/src/pages/modelServing/screens/global/DeleteInferenceServiceModal.tsx

```diff
@@ -34,62 +35,70 @@ const DeleteInferenceServiceModal: React.FC<DeleteInferenceServiceModalProps> =
     ? getDisplayNameFromK8sResource(inferenceService)
     : 'this deployed model';
 
+  const onDelete = async () => {
```

Comment:
Also -- maybe we consider the weirdness of NIM to be a utility you craft and import in. The general KServe flow does not need this extra logic of checking existing services count.

Reply:
I moved onDelete out of the Modal JSX to separation the logic from the component rendering. This way, onDelete can be defined and maintained as a standalone function, which might make it easier to test and modify independently.

Good point! I can create a utility specifically for the NIM logic, including the check for the existing services count, so that the core KServe flow remains unaffected.

Reply:
That's fair -- I'm not a fan of separating one-use functions from the JSX as you just put more in the way before seeing what the component returns... but that's fine I guess. I'm less concerned about new components you write, but reworking existing components feels like an unnecessary change imo -- and PRs should always try to stay on the path unless they are solving specific goals (refactoring, for instance). Anyways, it's all a nit, just explaining my reasoning; feel free to leave it separate. 

Thanks for the new function, I'll review the logic and get some testing going on here.